frontend/src/app/modules/intervention-form/service/intervention-form.service.ts

```diff
@@ -1,22 +1,41 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 
-import { Observable } from 'rxjs';
+import { from, Observable } from 'rxjs';
 
 import { InterventionFormData } from '@shared/domain/intervention.model';
-import { environment } from '@environment';
+import { InterventionsApiUrlsFactory } from '@shared/interventions-api-urls.factory';
+import { map, mergeAll } from 'rxjs/operators';
 
 @Injectable()
 export class InterventionFormService {
-  private interventionsUrl = environment.APIUrl + 'interventions';
-
   constructor(private http: HttpClient) {}
 
   post(formData: InterventionFormData): Observable<any> {
-    return this.http.post(this.interventionsUrl, formData);
+    return this.http.post(InterventionsApiUrlsFactory.interventions, formData);
   }
 
   update(formData: InterventionFormData): Observable<any> {
-    return this.http.put(`${this.interventionsUrl}/${formData.id}`, formData);
+    return this.http.put(InterventionsApiUrlsFactory.intervention(formData.id), formData);
+  }
+
+  /**
+   * Concurrent attachments upload method.
+   */
+  uploadAttachments(interventionId: string, attachments: File[]): Observable<any> {
+    return from(attachments).pipe(
+      map(attachment =>
+        this.http.post(
+          InterventionsApiUrlsFactory.attachments(interventionId),
+          attachment,
+          contentDispositionHeader(attachment.name),
+        ),
+      ),
+      mergeAll(3),
```

Comment:
nit: maybe better introduce descriptive constant for that number

Reply:
Sure, will do :)